setuptools/command/build_py.py

```diff
@@ -240,3 +251,13 @@ def assert_relative(path):
         % path
     )
     raise DistutilsSetupError(msg)
+
+
+def has_valid_modules(directory):
+    return any(
+        all(
+            part.isidentifier()
+            for part in potential_module.relative_to(directory).with_suffix("").parts
+        )
+        for potential_module in Path(directory).glob("**/*.py")
```

Comment:
Here I thought about checking against `importlib.machinery.all_suffixes`, but I decided to go only with `.py` because I don't know what to do with `.so` files (that are listed in `all_suffixes`... (Maybe the `.so` is not generated by setuptools, but instead compiled out-of-band and included by the user on purpose).